claat/main.go

```diff
@@ -61,36 +61,44 @@ func main() {
 
 	flag.Usage = usage
 	flag.CommandLine.Parse(os.Args[2:])
-	cmd.ExtraVars = cmd.ParseExtraVars(*extra)
+
+	var err error
+	cmd.ExtraVars, err = cmd.ParseExtraVars(*extra)
+	if err != nil {
+		os.Exit(1)
+	}
 
 	switch os.Args[1] {
 	case "export":
-		cmd.CmdExport(cmd.CmdExportOptions{
+		if ok := cmd.CmdExport(cmd.CmdExportOptions{
 			AuthToken: *authToken,
 			Expenv:    *expenv,
 			GlobalGA:  *globalGA,
 			Output:    *output,
 			Prefix:    *prefix,
 			Tmplout:   *tmplout,
-		})
+		}); !ok {
```

Comment:
Not sure I see the motivation for this change, in general. You avoided a custom function but in return have to add three if statements to catch the false return (and any new subcommands will need to check return as well) and you lost the error content. The previous implementation didn't require any error checking and displayed the error content for the user.

Reply:
Creating a custom implementation of a standard library function (that doesn't behave like the standard library function) seems like bigger problem to me. The error content is still displayed to the end user, just like errorf did.

Reply:
I think it's more idiomatic for the method to return the error(s) instead using something like multierror and then callers should check if the error is nil.

Reply:
Ordinarily I'd agree, except the information flow model for this code seems to be "cmd functions are responsible for their own user IO". If we're going to propagate error information up and use it there, we should do the same for non-error info. As the code stands now, there's no functionality benefit from having that error information available in main.go.

Unless we'd rather have the idiomaticism of just passing the error up and ignoring the content, depending solely on its presence for determining the exit code? I'll write a commit for that.

Reply:
I would think about it this way:

If CLI functions are managing their own IO, then they should accept some kind of formatter (or just use os.Stderr/out for now), and return an `int` representing the exit code:

```golang
func CmdUpdate(opts CmdUpdateOptions) int {
  resp, err := doTheThing()
  if err != nil {
    formatter.Errorf("something bad: %s", err)
    return 2
  }
  formatter.Info("this worked")
}
```

If CLI functions are pure go functions, then they should return their own data type as appropriate and an error. If it's a no-op, they should return only an error:

```golang
func CmdNoOutput(opts CmdNotOutputOptions) error {
  return doTheThing();
}

func CmdListThings(opts CmdNotListThingsOptions) ([]string, error) {
  list, err := doToThing();
  if err != nil {
    return nil, errwrap.Wrapf(err, "failed to do the thing")
  }
  return list, nil
}
```